src/cargo/sources/path.rs

```diff
@@ -498,9 +498,7 @@ impl<'cfg> Debug for PathSource<'cfg> {
 
 impl<'cfg> Source for PathSource<'cfg> {
     fn query(&mut self, dep: &Dependency, f: &mut dyn FnMut(Summary)) -> Poll<CargoResult<()>> {
-        if !self.updated {
-            return Poll::Pending;
-        }
```

Comment:
My understanding when I first read the code was that mutability had changed and so this was now possible, but the facts don't agree with me.
More fundamentally, there is no asynchronous work for a `PathSource` to do so we can just "do the update" instead of scheduling it for later.